File: src/Domain/Issue.php
<?php
declare(strict_types=1);

namespace Symphony\Domain;

class Issue
{
    public function __construct(
        public readonly string $id,
        public readonly string $identifier,
        public readonly string $title,
        public readonly ?string $description,
        public readonly ?int $priority,
        public readonly string $state,
        public readonly ?string $branchName,
        public readonly ?string $url,
        public readonly array $labels,
        public readonly array $blockedBy,
        public readonly ?\DateTimeImmutable $createdAt,
        public readonly ?\DateTimeImmutable $updatedAt,
    ) {}
}

File: internal/tracker/issue.go
package tracker

import "time"

// BlockerRef references a blocking issue.
type BlockerRef struct {
	ID         string
	Identifier string
	State      string
}

// Issue represents a tracker issue.
type Issue struct {
	ID          string
	Identifier  string
	Title       string
	Description string
	Priority    *int
	State       string
	BranchName  string
	URL         string
	Labels      []string
	BlockedBy   []BlockerRef
	CreatedAt   *time.Time
	UpdatedAt   *time.Time
}